### PR TITLE
Improve UML diagrams and developer documentation

### LINKED ISSUE #324: Evaluate/Improve UML diagrams
Can consider the following suggested UML changes (if makes sense)
Details: #304, #303, #281

### PR DESCRIPTION
**Pull request by kevin-pek**
Resolves #324.

Changes made:
- Remove UiPart class from `UiClassDiagram.puml`
- Split sequence diagram from `AssignSequenceDiagram.puml` into 2 separate diagrams and updated DG for better clarity.
- Removed fields from `ModelClassDiagram.puml`.
- Removed `BetterModelClassDiagraml.puml` since it is no longer used.
- Removed irrelevant labels from dependency relationships in `LogicClassDiagram.puml`.

### CODE REVIEW

**Comment by Jasperabez** [comment]
Diagrams:
UiClassDiagram
AssignParserSequenceDiagram
AssignSequenceDiagram
ModelClassDiagram
LogicClassDiagram

**Comment by kevin-pek** [comment]
Am thinking if its a good idea to replace with the conditional flow of the assign command to just show the `AssignCommand` calling its own `assignEventToVendor(m)` method? Sequence diagram might still be too complicated.

**Comment by Jasperabez** [comment]
@kevin-pek looks correct but maybe the line to the right of `EventDetailsPanel` in the UI diagram could be adjusted to the left if possible.

Would prob get @C5hives to double check the diagrams and approve too.

**Comment by C5hives** [comment]
For the `Model` class diagram, should the multiplicities for `Vendor` i.e. `* filtered` and `* all` be swapped? I think the current notation is incorrect.

**Comment by C5hives** [comment]
Also, I think keeping the labels `creates`, etc. helps a lot with the readability of the `Logic` class diagram.

Maybe consider re-adding them and just remove the `<<>>` brackets?

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ The sections below give more details of each component.
 
 The **API** of this component is specified in [`Ui.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/Ui.java)
 
-<puml src="diagrams/UiClassDiagram.puml" alt="Structure of the UI Component"/>
+<puml src="diagrams/UiClassDiagram.puml" alt="Structure of the UI Component" width="800" />
 
 The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
```
Comment (C5hives) [inline docs/DeveloperGuide.md:74]: Do you want to explain that the inheritance from `UiPart` was intentionally left out in the class diagram to reduce clutter?

**Comment by kevin-pek** [comment]
Screenshot 2024-11-11 at 12 06 24 AM

Have simplified the sequence diagram to this, thoughts?

**Comment by C5hives** [comment]
> Screenshot 2024-11-11 at 12 06 24 AM
> 
> Have simplified the sequence diagram to this, thoughts?

The entire alt block is jutting out quite a bit on the left. I assume this is PlantUml's doing again?

**Comment by kevin-pek** [comment]
> > Screenshot 2024-11-11 at 12 06 24 AM
> > Have simplified the sequence diagram to this, thoughts?
> 
> The entire alt block is jutting out quite a bit on the left. I assume this is PlantUml's doing again?
> 
> But it definitely looks better. I think this is the way to go

Issue is caused by the CommandException being thrown in the else case, not too sure how to better present this. 🤔

**Comment by kevin-pek** [comment]
> For the `Model` class diagram, should the multiplicities for `Vendor` i.e. `* filtered` and `* all` be swapped? I think the current notation is incorrect.

I believe the multiplicity should be correct? A single `UniqueList` or `ModelManager` should contain zero or more `Vendor` objects and so on.

**Comment by kevin-pek** [comment]
> Also, I think keeping the labels `creates`, etc. helps a lot with the readability of the `Logic` class diagram.
> 
> Maybe consider re-adding them and just remove the `<<>>` brackets?
> 
> Another thing we missed: For the `Logic` class diagram, the note for `XYZCommand` is technically wrong as we no longer have an `AddCommand`. Should be `CreateVendorCommand` or `CreateEventCommand`, or can be another command entirely.

Was thinking about this, but couldn't find examples in the textbook where dependency relationships (dotted arrows) included labels, so decided to remove them.

**Comment by C5hives** [comment]
> > For the `Model` class diagram, should the multiplicities for `Vendor` i.e. `* filtered` and `* all` be swapped? I think the current notation is incorrect.
> 
> I believe the multiplicity should be correct? A single `UniqueList` or `ModelManager` should contain zero or more `Vendor` objects and so on.

Error on my part. Was referring to the association roles (`filtered` vs. `all`).
image

Since `ModelManager` only holds on to a filtered `Vendor` list, the association roles in the current diagram would be wrong. (see how the roles are flipped for `Vendor` compared to `Event`)
image

Again, I think this is an issue with how PlantUml renders the association labels. The notation is correct in text form, but overlaps and causes confusion when rendered.
image

File: docs/diagrams/AssignSequenceDiagram.puml
Comment (C5hives) [inline docs/diagrams/AssignSequenceDiagram.puml:1]: I think this is serviceable. @AY2425S1-CS2103-F13-2/developers might want to chime in for this.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -125,6 +131,12 @@ How commands work:
 
 <puml src="diagrams/ModelClassDiagram.puml" width="640" />
 
+<box type="info" seamless>
+
+**Note** The classes representing the fields of the models `Vendor`, `Event` and `Association` were not included to reduce the clutter of the diagram.
```
Comment (C5hives) [inline docs/DeveloperGuide.md:136]: Consider rephrasing to:

"To reduce visual clutter, auxiliary classes within the `Vendor`, `Event` and `Association` classes have been omitted from the diagram."

**Comment by C5hives** [comment]
> > Also, I think keeping the labels `creates`, etc. helps a lot with the readability of the `Logic` class diagram.
> > Maybe consider re-adding them and just remove the `<<>>` brackets?
> > Another thing we missed: For the `Logic` class diagram, the note for `XYZCommand` is technically wrong as we no longer have an `AddCommand`. Should be `CreateVendorCommand` or `CreateEventCommand`, or can be another command entirely.
> 
> Was thinking about this, but couldn't find examples in the textbook where dependency relationships (dotted arrows) included labels, so decided to remove them to play it safe. Not sure if this is the right call, though?

I've tried cross-referencing other groups in our tutorial slot but they have all left the diagram untouched. I'm not sure whether omitting the labels is a good idea. @AY2425S1-CS2103-F13-2/developers thoughts on this?

**Comment by kevin-pek** [comment]
> > > Also, I think keeping the labels `creates`, etc. helps a lot with the readability of the `Logic` class diagram.
> > > Maybe consider re-adding them and just remove the `<<>>` brackets?
> > > Another thing we missed: For the `Logic` class diagram, the note for `XYZCommand` is technically wrong as we no longer have an `AddCommand`. Should be `CreateVendorCommand` or `CreateEventCommand`, or can be another command entirely.
> > 
> > 
> > Was thinking about this, but couldn't find examples in the textbook where dependency relationships (dotted arrows) included labels, so decided to remove them to play it safe. Not sure if this is the right call, though?
> 
> I've tried cross-referencing other groups in our tutorial slot but they have all left the diagram untouched. I'm not sure whether omitting the labels is a good idea. @AY2425S1-CS2103-F13-2/developers thoughts on this?

Have reverted the changes for now, maybe we can refer to past year forums to verify this.

File: docs/diagrams/AssignSequenceDiagram.puml
Comment (Jasperabez) [reply docs/diagrams/AssignSequenceDiagram.puml:1]: LGTM!

**Review by Jasperabez — CHANGES_REQUESTED** [review]
Would be good to standardize the lack of `<<>>` for ParserClasses UML too

**Comment by C5hives** [comment]
Current state of diagrams. Is this better?

`LogicClassDiagram.puml`: 
image
`ParserClasses.puml`: 
image
`ModelClassDiagram.puml`: 
image

**Review by Jasperabez — APPROVED** [review]
LGTM! Good work guys

**Pull request merged**